### ISSUE TITLE
Geom.label does not work with Date x-axis

### ISSUE THREAD

**Issue by mbauman** [dcjones/Gadfly.jl#501]
The Geom.label placement algorithm does lots of math with the x_continuous scale, which doesn't work very well for ordinal (but non-`Number`) types like Dates.

```julia
julia> using Dates, Gadfly
julia> plot(x=[Date(2014)], y=[1], label=["foo"], Geom.label)
ERROR: `*` has no method matching *(::Date, ::Measure{Float64,MeasureNil})
 in deferred_label_context at /Users/mbauman/.julia/v0.3/Gadfly/src/geom/label.jl:48
 in anonymous at /Users/mbauman/.julia/v0.3/Gadfly/src/geom/label.jl:265
 in realize at /Users/mbauman/.julia/v0.3/Compose/src/container.jl:313
 in drawpart at /Users/mbauman/.julia/v0.3/Compose/src/container.jl:437
 in draw at /Users/mbauman/.julia/v0.3/Compose/src/container.jl:394
 in display at /Users/mbauman/.julia/v0.3/TerminalExtensions/src/TerminalExtensions.jl:167
 in display at /Users/mbauman/.juliarc.jl:47
 in display at /Users/mbauman/.julia/v0.3/Gadfly/src/Gadfly.jl:149
 in print_response at REPL.jl:139
 in print_response at REPL.jl:124
 in anonymous at REPL.jl:586
 in anonymous at /Users/mbauman/.julia/v0.3/TerminalExtensions/src/TerminalExtensions.jl:225
 in run_interface at ./LineEdit.jl:1379
 in run_interface at /Users/mbauman/Code/julia-0.3/usr/lib/julia/sys.dylib
 in run_frontend at ./REPL.jl:818
 in run_repl at ./REPL.jl:169
 in _start at ./client.jl:399
 in _start at /Users/mbauman/Code/julia-0.3/usr/lib/julia/sys.dylib
```

**Comment by IainNZ** [comment]
I think this is the same as what I just reported #539 and maybe fixed in #540

**Comment by dcjones** [comment]
Fixed by @IainNZ in #540.

**Comment by mbauman** [comment]
Thanks Iain!
